x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -312,10 +318,13 @@ private static NamedExpression createEnrichFieldExpression(
         }
     }
 
-    private static class ResolveRefs extends BaseAnalyzerRule {
+    private static class ResolveRefs extends ParameterizedAnalyzerRule<LogicalPlan, AnalyzerContext> {
```

Comment:
Is this ok? I need the context.

Reply:
Better to use the approach for Enrich - create a separate rule that gets the analyzer context and partially/fully resolves the lookup since all information for it is already available.
If there are extra things, complete the resolution inside ResolveRefs (without having to pass the context)  - essentially the first rule does a first pass, letting ResolveRefs to fail the resolution if necessary.